api/requester/requester.go

```diff
@@ -75,139 +80,201 @@ func SetVerbose(verb bool) {
 	if userCfg == nil || reqCfg == nil {
-		return nil, errors.New("[ Send ] Configs must be initialized")
+		return nil, errors.New("[ SendWithSeed ] Configs must be initialized")
 	}
 
-	params, err := constructParams(reqCfg.Params)
-	if err != nil {
-		return nil, errors.Wrap(err, "[ Send ] Problem with serializing params")
-	}
+	ks := platformpolicy.NewKeyProcessor()
 
-	callerRef, err := insolar.NewReferenceFromBase58(userCfg.Caller)
+	pem, err := ks.ExportPublicKeyPEM(userCfg.privateKeyObject.(*ecdsa.PrivateKey).Public())
 	if err != nil {
-		return nil, errors.Wrap(err, "[ Send ] Failed to parse userCfg.Caller")
+		return nil, errors.Wrap(err, "[ SendWithSeed ] Cant export public key to PEM")
 	}
 
-	serRequest, err := insolar.MarshalArgs(
-		*callerRef,
-		reqCfg.Method,
-		params,
-		seed)
-	if err != nil {
-		return nil, errors.Wrap(err, "[ Send ] Problem with serializing request")
-	}
+	reqCfg.Params.Reference = userCfg.Caller
+	reqCfg.Params.PublicKey = string(pem)
+	reqCfg.Params.Seed = seed
 
 	verboseInfo(ctx, "Signing request ...")
-	cs := scheme.Signer(userCfg.privateKeyObject)
-	signature, err := cs.Sign(serRequest)
+	dataToSign, err := json.Marshal(reqCfg)
+	if err != nil {
+		return nil, errors.Wrap(err, "[ SendWithSeed ] Config request marshaling failed")
+	}
+	signature, err := sign(userCfg.privateKeyObject, dataToSign)
 	if err != nil {
-		return nil, errors.Wrap(err, "[ Send ] Problem with signing request")
+		return nil, errors.Wrap(err, "[ SendWithSeed ] Problem with signing request")
 	}
 	verboseInfo(ctx, "Signing request completed")
 
-	postParams := PostParams{
-		"params":    params,
-		"method":    reqCfg.Method,
-		"reference": userCfg.Caller,
-		"seed":      seed,
-		"signature": signature.Bytes(),
-	}
-	if reqCfg.LogLevel != nil {
-		postParams["logLevel"] = reqCfg.LogLevel
+	body, err := GetResponseBodyContract(url, *reqCfg, signature)
+	if err != nil {
+		return nil, errors.Wrap(err, "[ SendWithSeed ] Problem with sending target request")
 	}
 
-	body, err := GetResponseBody(url, postParams)
+	return body, nil
+}
 
+func sign(privateKey crypto.PrivateKey, data []byte) (string, error) {
+	hash := sha256.Sum256(data)
+
+	r, s, err := ecdsa.Sign(rand.Reader, privateKey.(*ecdsa.PrivateKey), hash[:])
 	if err != nil {
-		return nil, errors.Wrap(err, "[ Send ] Problem with sending target request")
+		return "", errors.Wrap(err, "[ sign ] Cant sign data")
 	}
 
-	return body, nil
+	return PointsToDER(r, s), nil
+}
+
+func PointsToDER(r, s *big.Int) string {
+	prefixPoint := func(b []byte) []byte {
+		if len(b) == 0 {
+			b = []byte{0x00}
+		}
+		if b[0]&0x80 != 0 {
+			paddedBytes := make([]byte, len(b)+1)
+			copy(paddedBytes[1:], b)
+			b = paddedBytes
+		}
+		return b
+	}
+
+	rb := prefixPoint(r.Bytes())
+	sb := prefixPoint(s.Bytes())
+
+	// DER encoding:
+	// 0x30 + z + 0x02 + len(rb) + rb + 0x02 + len(sb) + sb
```

Comment:
what is that?

Reply:
Its how we construct DER signature from two big numbers
![image](https://user-images.githubusercontent.com/2853815/59592702-3875a000-90f9-11e9-9403-018c03b8db8d.png)

Reply:
It should be as comment in code.